### PR TITLE
public auto parallel high level api

### PR DESCRIPTION
**Pull request by jeff41404**
### PR Category
User Experience

### PR Types
New features

### Description
pcard-88390
public auto parallel high level api: `paddle.distributed.to_distributed`
```Python
model = Model() # Model has no distributed code or markup in Model, temporarily supports certain models of transfromer type
input_seqs = np.random.randint(
    low=0, high=1024, size=(BATCH_SIZE * BATCH_NUM, SEQ_LENGTH)
).astype("int64")
labels = np.random.randint(
    low=0, high=1024, size=(BATCH_SIZE * BATCH_NUM, SEQ_LENGTH)
).astype("int64")
dataset = RandomDataset(
    input_seqs, labels, BATCH_SIZE * BATCH_NUM
)
sampler = paddle.io.BatchSampler(
    dataset, batch_size=BATCH_SIZE, shuffle=False, drop_last=True
)
loader = paddle.io.DataLoader(
    dataset, batch_sampler=sampler
)
opt = paddle.optimizer.SGD(
    learning_rate=0.1, parameters=model.parameters()
)
input_seq_spec = paddle.static.InputSpec(
    [BATCH_SIZE, SEQ_LENGTH], 'float32', 'input_seq', True
)
dist_config = ToDistributedConfig()
# dist_config.input_spec = [input_seq_spec]
dist_config.sequence_parallel = True

# wrap model, opt, dataloader by using **to_distributed**, use 8 devices in 1 node 
dist_model, dist_opt, dist_loader = to_distributed(
    model,
    opt,
    loader,
    device_num=8,
    node_num=1,
    config=dist_config,
)

# train as usual, dynamic graph
for epoch in range(EPOCHES):
    dist_model.train()
    for i, data in enumerate(dist_loader()):
        inputs, labels = data
        loss, _ = dist_model(inputs, labels=labels)
        print(f"epoch {epoch}, step {i}: loss {loss}")
        loss.backward()
        dist_opt.step()
        dist_opt.clear_grad()
```

### CODE REVIEW

**Comment by sunzhongkai588** [comment]
除代码示例外，似乎类型注释也报错，辛苦 @SigureMo review 一下
image

File: python/paddle/distributed/auto_parallel/high_level_api.py [hunk 1]
```diff
@@ -290,15 +297,15 @@ def to_distributed(
...
+            >>> # doctest: +SKIP('run in distributed env')
             >>> import math
             >>> import numpy as np
             >>> import paddle
             >>> import paddle.nn.functional as F
             >>> from paddle import nn
-            >>> from paddle.distributed.auto_parallel.high_level_api import (
-            >>>     ToDistributedConfig,
-            >>>     to_distributed,
-            >>> )
```
Comment (SigureMo) [inline python/paddle/distributed/auto_parallel/high_level_api.py:301]: > 除代码示例外，似乎类型注释也报错，辛苦 @SigureMo review 一下

312 行可以加一下 `# type: ignore[type-arg]`

```
>>> class RandomDataset(paddle.io.Dataset): # type: ignore[type-arg]
```

**Review by XieYunshen — APPROVED** [review]

**Review by zhiqiu — APPROVED** [review]
LGTM

File: python/paddle/distributed/auto_parallel/high_level_api.py (hunk 1 above)
Comment (jeff41404) [reply python/paddle/distributed/auto_parallel/high_level_api.py:301]: thanks, done

**Review by SigureMo — APPROVED** [review]
LGTMeow  for new API annotations

**Review by sunzhongkai588 — APPROVED** [review]
LGTM for docs

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**